python/prophet/plot.py

```diff
@@ -89,7 +89,13 @@ def plot(
     ax.set_ylabel(ylabel)
     if include_legend:
         ax.legend()
-    fig.tight_layout()
+    # Check if user passed in an ax
+    if ax is None:
+        fig.tight_layout()
+    else:
+        # Check fig.get_tight_layout() before calling fig.tight_layout()
+        if fig.get_tight_layout():
```

Comment:
Could I confirm what this is meant to check? I'm also happy to call tight_layout() only when the user hasn't provided an ax.

Reply:
I made changes based on your feedback. The changes include:

1. Tracking whether the ax was provided by the user or not. This is done by creating a new variable user_provided_ax at the start of the function.

2. Only calling fig.tight_layout() if the ax was not provided by the user. This is done by wrapping the fig.tight_layout() call in an if-statement that checks the user_provided_ax variable.